### PR TITLE
ScalarBars.add_scalar_bar: correct annotation text scaling setting, when fixed font size is given

### PR DESCRIPTION
**Pull request by beppo-dd**
### Overview

Even if a label font size was given, size of `below_label` and `above_label` was scaled with the view port.

### Details

- Line 423 must be `SetAnnotationTextScalingOff` (value 0) for no scaling, see <https://vtk.org/doc/nightly/html/classvtkScalarBarActor.html#acfe97b358b89cd544df5e7826d59e4f7>
- Tested visually.

### CODE REVIEW

File: pyvista/plotting/scalar_bars.py [hunk 1]
```diff
@@ -421,6 +421,8 @@ def add_scalar_bar(self, title='', mapper=None, n_labels=5, italic=False,
         if label_font_size is not None or title_font_size is not None:
             scalar_bar.UnconstrainedFontSizeOn()
             scalar_bar.AnnotationTextScalingOff()
+        else:
+            scalar_bar.AnnotationTextScalingOn()
```
[the remark this answers is not in the document]
Comment (tkoyama010) [reply pyvista/plotting/scalar_bars.py:425]: @beppo-dd Sorry for late reply. I do not know the correct specifications. How do @akaszynski  think?

File: tests/plotting/test_plotting.py [hunk 2]
```diff
@@ -1576,6 +1576,18 @@ def test_user_annotations_scalar_bar_mesh(uniform):
...
+def test_fixed_font_size_annotation_text_scaling_off():
+    p = pyvista.Plotter()
+    sargs = {
+        'title_font_size': 12,
+        'label_font_size': 10}
+    p.add_mesh(examples.load_uniform(), clim=[100, 500], cmap='viridis',
+               below_color='blue', above_color='red',
+               annotations={300.: 'yum'},
+               scalar_bar_args=sargs)
+    p.show(before_close_callback=verify_cache_image)
```
[the remark this answers is not in the document]
Comment (tkoyama010) [reply tests/plotting/test_plotting.py:1588]: @beppo-dd Thanks. Could you add `image_cache/fixed_font_size_annotation_text_scaling_off.png` to Pull Request? It is need it to test CI and you can create it running
```bash
pytest test_plotting.py --reset_image_cache
```

[the remark this answers is not in the document]
Comment (tkoyama010) [reply tests/plotting/test_plotting.py:1588]: @beppo-dd This is the guide to setup guide for pyvista (maybe not for virtual linux).
https://dev.pyvista.org/extras/developer_notes.html#testing
The error says that your python version is not 3.8 or your machine is not x64. What is your python version and your machine environment?

[the remark this answers is not in the document]
Comment (beppo-dd) [reply tests/plotting/test_plotting.py:1588]: @tkoyama010 When I run your proposed command
> ```shell
> pytest test_plotting.py --reset_image_cache
> ```
I got the same error
```
tests/plotting/test_plotting.py Fatal Python error: Segmentation fault

Thread 0x00007fe57c080700 (most recent call first):
  File "/usr/lib/python3.8/threading.py", line 306 in wait
  File "/usr/lib/python3.8/threading.py", line 558 in wait
  File "/usr/lib/python3.8/threading.py", line 1252 in run
  File "/usr/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 890 in _bootstrap
```
My Linux system is a Ubuntu 20.04, 64 bit (x86_64, virtual, 3.8 GiB RAM), with Python 3.8.10 (Packages: pyvista 0.32.dev0, vtk 9.0.3, numpy 1.21.3, imageio 2.10.1, appdirs 1.4.4, scooby 0.5.7, meshio 4.4.6, matplotlib 3.4.3). I used already <https://dev.pyvista.org/extras/developer_notes.html#testing> to setup this system. Can you give me a hint for a working testing setup? Thanks.

[the remark this answers is not in the document]
Comment (beppo-dd) [reply tests/plotting/test_plotting.py:1588]: > The error says that your python version is not 3.8 or your machine is not x64.
@tkoyama010 I think, my error report was a bit confusing. The corresponding error occurs, when I run [`act`](https://github.com/nektos/act#readme), which runs the GitHub Actions, locally in a docker container. This is defined in [.github/workflows/testing-and-deployment.yml](https://github.com/pyvista/pyvista/blob/main/.github/workflows/testing-and-deployment.yml). The step "Set up Python" does not work in my act-setup.

[the remark this answers is not in the document]
Comment (tkoyama010) [reply tests/plotting/test_plotting.py:1588]: @beppo-dd  OK. I will reproduce it.(I didn't know the [act](https://github.com/nektos/act#readme). It seems useful! :))

[the remark this answers is not in the document]
Comment (beppo-dd) [reply tests/plotting/test_plotting.py:1588]: @tkoyama010 Thank you for your helpfulness. Meanwhile, I setup a new Ubuntu system on a hetzner cloud server and everything works perfect! So a could run the upper command and create the new image `image_cache/fixed_font_size_annotation_text_scaling_off.png`, see 681c772.

File: pyvista/plotting/scalar_bars.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (beppo-dd) [reply pyvista/plotting/scalar_bars.py:425]: @akaszynski I would be nice, if you could check the last question about the annotation text scaling.

[the remark this answers is not in the document]
Comment (akaszynski) [reply pyvista/plotting/scalar_bars.py:427]: This does not change the image regression (and it would be picked up as the thresholds are quite low).

**Review by akaszynski — APPROVED** [review]
Tested locally and agree with the behavior change. Thanks for adding this!

**Pull request merged**